src/krylov/pcg.c

```diff
@@ -540,18 +585,59 @@ hypre_PCGSolve( void *pcg_vdata,
          break;
       }
       alpha = gamma / sdotp;
-      if (! (alpha > HYPRE_REAL_MIN) )
+      /* alpha should always be greater zero for spd A, spd precond. and nonzero p, r*/
+      /* alpha = <r, Cr> / <p, Ap> */
+      /* for alpha close to 0 x and r will not change much unless x and/or r are close to 0 */
+      if (alpha <=0.0)
+      {
+         if (print_level > 1 && my_id == 0)
+	 {
+            hypre_printf("alpha %e", alpha);
+	 }
+         hypre_error_w_msg(HYPRE_ERROR_CONV, "Negative or zero alpha value in PCG");
+         if (skip_break == 0)
```

Comment:
Maybe I'm not reading this right, but shouldn't this break for `skip_break` equal `0` or `1`?  I'm looking at the documentation in `HYPRE_krylov.h`.

Reply:
Here we only check for alpha <= 0, however you are right that there is one more branch missing, where skip_break should be 1, the one where 0 < alpha < HYPRE_REAL_TRUE_MIN, so I will add that.

Reply:
Actually what I just said is not correct. I think the problem is just that the documentation is wrong. We need to shift the parameters by 1 down, and make the default 2, right now 0 and 3 are the same. If we do this things should be fine.